lib/active_model/serializer/adapter/json.rb

```diff
@@ -19,13 +19,17 @@ def serializable_hash(options = {})
                 array_serializer = association
                 @hash[name] = array_serializer.map do |item|
                   cache_check(item) do
-                    item.attributes(opts)
+                    opts[:include_nested_associations] ?
+                      self.class.new(item).serializable_hash :
+                      item.attributes(opts)
```

Comment:
This might be a good time to extract `add_association` to its own method, like the json_api adapter has